tests/test_valle_oganov.py

```diff
@@ -77,6 +77,10 @@ def test_derivatives_numerical():
     assert_derivatives(valle_oganov(), "numerical", False)
 
 
```

Comment:
Could you add the simplest possible test that checks that the correct neighbourhood is used when using `smooth_cutoff`? Ideally something that would have failed before this MR, but passes after your changes.

Reply:
I added a test with an atomic system that has three atoms that form a 90 degree angle. The cutoff is set so that both edges connected to the 90 degree corner are shorter than the cutoff but the third edge is longer. The result is that the 90 degree corner contributes to the representation, while the other two don't. The plot below shows that before the bug fix, the peak of the representation function was incorrectly at 180 degrees, while after the fix the peak is at 90 degrees as it should. The test simply checks that the peak is at 90 degrees.
![smooth_cutoff_fix](https://user-images.githubusercontent.com/29092449/236454398-97eeaaf1-6f09-4471-b9b4-cc8c182918ec.png)